omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyResource2_0.java

```diff
@@ -0,0 +1,246 @@
+	 *      java.lang.Object)
+	 */
+	@Override
+	public void setParent(Allergy instance, Patient patient) {
+		instance.setPatient(patient);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.SubResource#doGetAll(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public NeedsPaging<Allergy> doGetAll(Patient parent, RequestContext context) throws ResponseException {
+		List<Allergy> allergiesList = new ArrayList<Allergy>();
+		Allergies allergies = new Allergies();
+		if (parent != null) {
+			allergies = Context.getPatientService().getAllergies(parent);
+			if (allergies.getAllergyStatus().equals(Allergies.UNKNOWN)) {
+				// return 4040
+				throw new ObjectNotFoundException();
+			} else if (allergies.getAllergyStatus().equals(Allergies.NO_KNOWN_ALLERGIES)) {
+				// empty list
+			} else if (allergies.getAllergyStatus().equals(Allergies.SEE_LIST)) {
+				// return list
+				allergiesList.addAll(allergies);
+			}
+		}
+		return new NeedsPaging<Allergy>(allergiesList, context);
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public Allergy getByUniqueId(String uuid) {
+		try {
+			int allergyId = Integer.parseInt(uuid);
+			return Context.getPatientService().getAllergy(allergyId);
+		} catch (NumberFormatException  ex)
+		{
+			throw new ObjectNotFoundException();
+		}
+	}
+
+
+	private void delete(Allergy allergy, String reason){
+		Context.getPatientService().voidAllergy(allergy, reason);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#delete(java.lang.Object,
+	 *      java.lang.String, org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void delete(Allergy allergy, String reason, RequestContext context) throws ResponseException {
+		if (allergy != null) {			
+			delete(allergy, reason); // specific allergy
+		}
+		
+	}
+	
+	@Override
+	public void delete(String parentUniqueId, String uuid, String reason, RequestContext context) throws ResponseException {
+		if (parentUniqueId != null && uuid != null) {
+			// specific allergy with parent
+			Allergy allergy = getByUniqueId(uuid);
+			Patient patient = Context.getPatientService().getPatientByUuid(parentUniqueId);
+			setParent(allergy, patient);
+			delete(allergy, reason, context);
+		} else if (parentUniqueId != null && uuid == null) {
+			// all allergies
+			Patient patient = Context.getPatientService().getPatientByUuid(parentUniqueId);
+			Allergies allergies = new Allergies();
+			Context.getPatientService().setAllergies(patient, allergies);
+		} else if (uuid != null) {
+			// specific allergy
+			Allergy allergy = getByUniqueId(uuid);
+			delete(allergy, reason);
+		}
+	}
+
+	private void purge(Allergy newAllergy){
+		// api does not have purge method
+		Context.getPatientService().voidAllergy(newAllergy, null);
```

Comment:
if it can't be purged, then throw a ResourceDoesNotSupportOperationException. Don't change things and do a different operation that the client didn't request.